### ISSUE TITLE
_setup() method Enhancements

### ISSUE THREAD

**Issue by JamalAkram89** [oracle-samples/ofs-plugin-core#17]
1. “_setup” method currently gets called immediately when the plugin class is initiated. Implementing class may need perform additional setup/configs before initiating communication with OFS. This function should be made public and callable outside the constructor by implementing class (not forced to be called from constructor). 
2. Since the “_setup” method is private and called immediately by constructor, it takes away the ability from the implementing plugin to send following parameters to OFS while initializing.
     -     enableBackButton
     -     showHeader
     -     sendMessageAsJsObject
    This method can accept the following input parameters with defaults and form the ready Message accordingly.
     -     sendInitData: boolean
     -     enableBackButton: boolean
     -     showHeader: boolean
     -     sendMessageAsJsObject: boolean
     -     dataItems: Array<string>

Example:
```
Setup(sendInitData: boolean = true, enableBackButton: boolean = true, showHeader: boolean = true, sendMessageAsJsObject:boolean = false) {
       console.log("OFS plugin ready");
       window.addEventListener(
                "message",
                this._getWebMessage.bind(this),
                false
       );
       var messageData: OFSMessage = {
            apiVersion: 1,
            method: "ready",
            sendInitData: sendInitData,
            enableBackButton: enableBackButton,
            showHeader: showHeader,
            sendMessageAsJsObject: sendMessageAsJsObject
       };
       this._sendWebMessage(messageData);
}
```

**Comment by mailtodanish** [comment]
The plugin has been implemented according to the OFSC documentation. However, suggested enhancements to the plugin will not work after deployment in the OFSC.

**Comment by JamalAkram89** [comment]
> The plugin has been implemented according to the OFSC documentation. However, suggested enhancements to the plugin will not work after deployment in the OFSC.

Hi @mailtodanish , I might be missing something as i fail to see how it will not work after deployment in the OFSC. Could you explain a little please.

**Comment by mailtodanish** [comment]
@JamalAkram89 You are right. I encountered 'sendMessageAsJsObject' for the first time, so I was confused. Please ignore.
